components/textbox/textbox.tsx

```diff
@@ -66,257 +67,277 @@ type Props = {
+                        priorityProfiles,
                     });
                 }
             }
         }
 
-        if (this.props.channelId !== prevProps.channelId ||
-            this.props.currentTeamId !== prevProps.currentTeamId ||
-            this.props.rootId !== prevProps.rootId) {
+        if (channelId !== prevProps.channelId ||
+            currentTeamId !== prevProps.currentTeamId ||
+            rootId !== prevProps.rootId) {
             // Update channel id for CommandProvider and AppCommandProvider.
-            for (const provider of this.suggestionProviders) {
+            for (const provider of suggestionProviders) {
                 if (provider instanceof CommandProvider) {
                     provider.setProps({
-                        teamId: this.props.currentTeamId,
-                        channelId: this.props.channelId,
-                        rootId: this.props.rootId,
+                        teamId: currentTeamId,
+                        channelId,
+                        rootId,
                     });
                 }
                 if (provider instanceof AppCommandProvider) {
                     provider.setProps({
-                        teamId: this.props.currentTeamId,
-                        channelId: this.props.channelId,
-                        rootId: this.props.rootId,
+                        teamId: currentTeamId,
+                        channelId,
+                        rootId,
                     });
                 }
             }
         }
 
-        if (prevProps.value !== this.props.value) {
-            this.checkMessageLength(this.props.value);
-        }
-    }
-
-    componentDidUpdate(prevProps: Props) {
-        if (!prevProps.preview && this.props.preview) {
-            this.preview.current?.focus();
+        if (prevProps.value !== value) {
+            checkMessageLength(value);
         }
+    }, [actions, autocompleteGroups, channelId, currentTeamId, currentUserId, priorityProfiles, rootId, suggestionProviders, useChannelMentions, value, checkMessageLength]);
 
-        this.updateSuggestions(prevProps);
-    }
-
-    checkMessageLength = (message: string) => {
-        if (this.props.handlePostError) {
-            if (message.length > this.props.characterLimit) {
-                const errorMessage = (
-                    <FormattedMessage
-                        id='create_post.error_message'
-                        defaultMessage='Your message is too long. Character count: {length}/{limit}'
-                        values={{
-                            length: message.length,
-                            limit: this.props.characterLimit,
-                        }}
-                    />);
-                this.props.handlePostError(errorMessage);
-            } else {
-                this.props.handlePostError(null);
-            }
-        }
-    }
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => onChange(e);
```

Comment:
For these event handlers where we're just passing the raw event to props, we can just pass Textbox's props directly to the child component instead of wrapping them in `handleX` functions